tee-worker/omni-executor/evm-aggregators-client/src/kyber_client/types.rs

```diff
@@ -0,0 +1,225 @@
+	pub charge_fee_by: String,
+	#[serde(rename = "isInBps")]
+	pub is_in_bps: bool,
+	#[serde(rename = "feeReceiver")]
+	pub fee_receiver: String,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct RouteItem {
+	pub pool: String,
+	#[serde(rename = "tokenIn")]
+	pub token_in: String,
+	#[serde(rename = "tokenOut")]
+	pub token_out: String,
+	#[serde(rename = "limitReturnAmount")]
+	pub limit_return_amount: String,
+	#[serde(rename = "swapAmount")]
+	pub swap_amount: String,
+	#[serde(rename = "amountOut")]
+	pub amount_out: String,
+	pub exchange: String,
+	#[serde(rename = "poolLength")]
+	pub pool_length: i32,
+	#[serde(rename = "poolType")]
+	pub pool_type: String,
+	#[serde(rename = "poolExtra")]
+	pub pool_extra: PoolExtra,
+	pub extra: Value,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct PoolExtra {
+	#[serde(rename = "blockNumber")]
+	pub block_number: i32,
+	#[serde(rename = "priceLimit")]
+	pub price_limit: String, // or a BigUint if using a crate like `num`
+}
+
+#[derive(Debug, Serialize, Deserialize, Default)]
+pub struct SwapRequest {
+	#[serde(rename = "routeSummary")]
+	pub route_summary: RouteSummary,
+
+	#[serde(skip_serializing_if = "Option::is_none")]
+	pub sender: Option<String>,
+	#[serde(skip_serializing_if = "Option::is_none")]
+	pub recipient: Option<String>,
+	#[serde(skip_serializing_if = "Option::is_none")]
+	pub deadline: Option<i64>,
+	#[serde(rename = "slippageTolerance", skip_serializing_if = "Option::is_none")]
+	pub slippage_bps: Option<i64>,
+	#[serde(skip_serializing_if = "Option::is_none")]
+	pub referral: Option<String>,
+	#[serde(skip_serializing_if = "Option::is_none")]
+	pub source: Option<String>,
+	#[serde(rename = "enableGasEstimation", skip_serializing_if = "Option::is_none")]
+	pub enable_gas_estimation: Option<bool>,
+	#[serde(skip_serializing_if = "Option::is_none")]
+	pub permit: Option<String>,
+	#[serde(rename = "ignoreCappedSlippage", skip_serializing_if = "Option::is_none")]
+	pub ignore_capped_slippage: Option<bool>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct SwapResponse {
+	#[serde(rename = "amountIn")]
+	pub amount_in: String,
+	#[serde(rename = "amountInUsd")]
+	pub amount_in_usd: String,
+	#[serde(rename = "amountOut")]
+	pub amount_out: String,
+	#[serde(rename = "amountOutUsd")]
+	pub amount_out_usd: String,
+	pub gas: String,
+	#[serde(rename = "gasUsd")]
+	pub gas_usd: String,
+	#[serde(rename = "additionalCostUsd")]
+	pub additional_cost_usd: String,
+	#[serde(rename = "additionalCostMessage")]
+	pub additional_cost_message: String,
+	#[serde(rename = "outputChange")]
+	pub output_change: OutputChange,
+	pub data: String,
+	#[serde(rename = "routerAddress")]
+	pub router_address: String,
+	#[serde(rename = "transactionValue")]
+	pub transaction_value: String,
+}
+
+impl SwapResponse {
+	pub fn get_transaction_data(self) -> Result<(Vec<u8>, Address, U256, u64), ()> {
+		let data = hex::decode(&self.data).map_err(|e| {
```

Comment:
Should we inspect `data` to make sure at least that `in` and `out` tokens 'ca` match expectations ?